File: KoRadio/KoRadio.Model/Freelancer.cs
using KoRadio.Model.DTOs;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;


namespace KoRadio.Model
{

    public  class Freelancer
    {
		public int FreelancerId { get; set; }

		public string Bio { get; set; } = null!;

		public decimal Rating { get; set; }

		public int ExperianceYears { get; set; }

		public List<DayOfWeek> WorkingDays { get; set; }

		public TimeOnly StartTime { get; set; }

		public TimeOnly EndTime { get; set; }

		public bool IsDeleted { get; set; }
		public bool IsApplicant { get; set; }

		public int? TotalRatings { get; set; }

		public double? RatingSum { get; set; }
		[NotMapped]
		public double? AverageRating => TotalRatings == 0 ? 0 : RatingSum / TotalRatings;

		public byte[]? CV { get; set; }


		public virtual User FreelancerNavigation { get; set; } = null!;

		public virtual ICollection<FreelancerService> FreelancerServices { get; set; } = new List<FreelancerService>();

		
	}
}

File: KoRadio/UI/ko_radio_desktop/lib/models/freelancer.dart


import 'package:json_annotation/json_annotation.dart';
import 'package:ko_radio_desktop/models/freelancer_service.dart';
import 'package:ko_radio_desktop/models/location.dart';
import 'package:ko_radio_desktop/models/user.dart';

part 'freelancer.g.dart';
@JsonSerializable()
class Freelancer{
  int freelancerId;
  String bio;
  double rating;
  int experianceYears;
  List<String>? workingDays;
  String startTime;
  String endTime;
  List<FreelancerService> freelancerServices;
  User? freelancerNavigation;
  bool isApplicant;
  bool isDeleted;



  
  Freelancer({required this.bio,  required this.freelancerServices, required this.freelancerNavigation,
   required this.isApplicant, required this.isDeleted, required this.rating, required this.experianceYears, required this.workingDays, required this.startTime, required this.endTime,
   required this.freelancerId });

  factory Freelancer.fromJson(Map<String, dynamic> json) => _$FreelancerFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$FreelancerToJson(this);
}
